### PR TITLE
CRIMAPP-1435 validate decisions against schema

### PR DESCRIPTION
**Pull request by timpeat**
## Description of change

Validate decision against json schema.

## Link to relevant ticket
https://dsdmoj.atlassian.net/browse/CRIMAPP-1435

## Notes for reviewer / how to test

Uses the same schema version as the original application.

### CODE REVIEW

File: app/api/datastore/base.rb [hunk 1]
```diff
@@ -12,6 +12,8 @@ class Base < Grape::API
     end
 
     rescue_from LaaCrimeSchemas::Errors::ValidationError do |ex|
+      Rails.error.report(ex, handled: true)
+
```
Comment (timpeat) [inline app/api/datastore/base.rb:16]: Schema errors were not being reported. I think the client should probably be doing the reporting of these but I've added so we get flagged for the time being.

File: app/services/operations/complete_application.rb [hunk 2]
```diff
@@ -3,8 +3,8 @@ class CompleteApplication
     def initialize(application_id:, decisions:)
       @application = CrimeApplication.find(application_id)
       @decisions = decisions
-      # TODO: revert after user research
```
Comment (timpeat) [inline app/services/operations/complete_application.rb:6]: Re-enable as this now works.

File: app/services/operations/complete_application.rb [hunk 3]
```diff
@@ -24,14 +24,26 @@ def call # rubocop:disable Metrics/AbcSize
 
     private
 
-    # TODO: revert after user research
```
Comment (timpeat) [inline app/services/operations/complete_application.rb:27]: I could not get the original implementation of this to work. There seemed to be a caching issue which caused flaky specs.

File: config/environments/test.rb [hunk 4]
```diff
@@ -28,7 +28,7 @@
   config.cache_store = :null_store
 
   # Raise exceptions instead of rendering exception templates.
-  config.action_dispatch.show_exceptions = false
```
Comment (timpeat) [inline config/environments/test.rb:31]: fixes deprecation warning

**Review by hiboabd — APPROVED** [review]

**Pull request merged**